src/journal/journald.c

```diff
@@ -52,7 +52,7 @@ int main(int argc, char *argv[]) {
                 goto finish;
 
         server_vacuum(&server, false);
-        server_flush_to_var(&server);
+        server_flush_to_var(&server, true);
```

Comment:
I've just tested steps from the #1937 (with reverted https://github.com/systemd/systemd/pull/2057/commits/6b95e47067e52be60b37307d57f1ed2b5419b9d5)
```
-bash-4.3# ls -ld /var/log/journal/
drwxr-sr-x+ 3 root systemd-journal 1024 Dec 16 06:10 /var/log/journal/

-bash-4.3# ls -ld /var/log/journal/5d7274cd96864633b24c56ae4a838916/
drwxr-sr-x+ 2 root systemd-journal 1024 Dec 16 06:10 /var/log/journal/5d7274cd96864633b24c56ae4a838916/

-bash-4.3 ls -ld /var/log/journal/5d7274cd96864633b24c56ae4a838916/system.journa
-rw-r-----+ 1 root root 4845568 Dec 16 06:13 /var/log/journal/5d7274cd96864633b24c56ae4a838916/system.journal
```
The race is here. Am I missing something?